src/Xamarin.Android.Build.Tasks/Tasks/DetermineJavaLibrariesToCompile.cs

```diff
@@ -62,18 +63,40 @@ public override bool Execute ()
 					jars.Add (jar);
 
 			var distinct  = MonoAndroidHelper.DistinctFilesByContent (jars);
-			jars          = jars.Where (j => distinct.Contains (j)).ToList ();
 
-			JavaLibrariesToCompile = jars.Where (j => !IsExcluded (j.ItemSpec)).ToArray ();
-			ReferenceJavaLibraries = (ExternalJavaLibraries ?? Enumerable.Empty<ITaskItem> ())
-				.Concat (jars.Except (JavaLibrariesToCompile)).ToArray ();
+			var javaLibrariesToCompile = new List<ITaskItem> ();
+			var referenceJavaLibraries = new List<ITaskItem> (ExternalJavaLibraries);
+
+			foreach (var item in distinct) {
+				if (!HasClassFiles (item.ItemSpec))
+					continue;
+				if (IsExcluded (item.ItemSpec)) {
+					referenceJavaLibraries.Add (item);
+				} else {
+					javaLibrariesToCompile.Add (item);
+				}
+			}
+			JavaLibrariesToCompile = javaLibrariesToCompile.ToArray ();
+			ReferenceJavaLibraries = referenceJavaLibraries.ToArray ();
+			//jars          = jars.Where (j => distinct.Contains (j)).ToList ();
+
+			//JavaLibrariesToCompile = jars.Where (j => !IsExcluded (j.ItemSpec)).ToArray ();
+			//ReferenceJavaLibraries = (ExternalJavaLibraries ?? Enumerable.Empty<ITaskItem> ())
+			//	.Concat (jars.Except (JavaLibrariesToCompile)).ToArray ();
 
 			Log.LogDebugTaskItems ("  JavaLibrariesToCompile:", JavaLibrariesToCompile);
 			Log.LogDebugTaskItems ("  ReferenceJavaLibraries:", ReferenceJavaLibraries);
 
 			return true;
 		}
 
+		bool HasClassFiles (string jar)
+		{
+			return Files.ZipContains (jar, (zip)=> {
+				return zip.Any (x => Path.GetExtension (x.FullName) == ".class");
```

Comment:
Considering that `Path` is designed to work with operating system paths, while the `.zip` file *might not* be an operating system path, it might be better to instead use:

```csharp
return zip.Any (x => x.EndsWith (".class", StringComparison.OrdinalIgnoreCase));
```

Then there's the case sensitivity issue. :-)